blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -5,6 +5,7 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.net.Socket;
+import java.nio.ByteBuffer;
```

Comment:
Maybe but it seems very wasteful (and inefficient) copying all the data into one array and then copying it into another before sending. So I would prefer one of the following three options:

1. Copy the data into one array and send that down the OutputStream in  a single synchronized write. I realise that two tests failed with this approach. I wonder if there is any befefit in debugging why).
2. Go with your original approach and perform the output in multiple writes and enclose them in a synchronized block.
3. Use a SocketChannel and do not use synchronization (I realise this is a larger change so probably isn't an option at the moment).

@tomjenkinson do you have an opinion on which option to implement - this is an often used routine so it needs to perform well.